lms/validation/_api.py

```diff
@@ -84,10 +84,11 @@ class APICreateAssignmentSchema(PyramidRequestSchema):
 
     class Content(Schema):
```

Comment:
I think the validation here is a bit lacking with this changes. Some fields are required only in one LMS but not other. This could be  enforced on `@validates_schema`but not sure what the better alternatives are.

- Split the view per LMS? (canvas vs the rest) and use one schema in each view

- Use some sort of polymorphism with marshemllow. This doesn't seem to be supported by the core library  https://github.com/marshmallow-code/apispec/pull/182 and will need to use some other dependency like https://github.com/marshmallow-code/marshmallow-oneofschema

Reply:
Yeah, I think splitting the views might be best.

We could have it post to one URL in Canvas and to another URL in non-Canvas, since the backend uses the `JSConfig` to control what URL the frontend calls. Then use different views for different routes. Alternatively we could keep the URL the same but use the `request_param` predicate in `@view_config` to make it use one schema or another depending on a request param.

Either way you could either use two different views or one view with two `@view_config`'s on it:

```python
@view_config(..., schema=APICreateAssignmentSchema)
@view_config(..., schema=APICanvasCreateAssignmentSchema)
def create_assignment(request):
    ...
```

Whatever works best